packages/yourloops/test/integration/assert/caregivers-list.assert.ts

```diff
@@ -0,0 +1,171 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { screen, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { mockCaregiverUser } from '../mock/direct-share.api.mock'
+import NotificationApi from '../../../lib/notifications/notification.api'
+import { NotificationType } from '../../../lib/notifications/models/enums/notification-type.enum'
+import { type Notification } from '../../../lib/notifications/models/notification.model'
+import DirectShareApi, { PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_MESSAGE } from '../../../lib/share/direct-share.api'
+import { UserInviteStatus } from '../../../lib/team/models/enums/user-invite-status.enum'
+
+export const checkCaregiversListLayout = async () => {
+  expect(await screen.findByTestId('patient-caregivers-list')).toBeVisible()
+
+  const addCaregiverButton = await screen.findByTestId('add-caregiver-button')
+  expect(addCaregiverButton).toBeVisible()
+  expect(addCaregiverButton).toBeEnabled()
+}
+
+export const checkAddRemoveCaregiver = async () => {
+  const caregiversTable = screen.getByTestId('patient-caregivers-list')
+  expect(caregiversTable).toHaveTextContent('Last nameFirst nameEmail')
+
+  const caregiverRow = within(caregiversTable).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}`)
+  expect(caregiverRow).toBeVisible()
+  expect(caregiverRow).toHaveTextContent('UserCaregivercaregiver@mail.com')
+  expect(within(caregiverRow).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}-button-remove`)).toBeVisible()
+
+  const addCaregiverButton = await screen.findByTestId('add-caregiver-button')
+  await userEvent.click(addCaregiverButton)
+
+  const addCaregiverDialog = screen.getByTestId('patient-add-caregiver-dialog')
+  expect(addCaregiverDialog).toBeVisible()
+  expect(within(addCaregiverDialog).getByText('Add a caregiver')).toBeVisible()
+  const emailInput = within(addCaregiverDialog).getByRole('textbox', { name: 'Email' })
+  expect(emailInput).toBeVisible()
+
+  const newCaregiverEmail = 'new-caregiver@mail.com'
+  await userEvent.type(emailInput, 'new-caregiver@mail.com')
+  const inviteButton = within(addCaregiverDialog).getByRole('button', { name: 'Invite' })
+  expect(inviteButton).toBeEnabled()
+
+  jest.spyOn(NotificationApi, 'getSentInvitations').mockResolvedValueOnce([{
+    email: newCaregiverEmail,
+    type: NotificationType.directInvitation,
+    target: { id: 'target-id' },
+    id: 'my-id'
+  } as Notification])
+  await userEvent.click(inviteButton)
+
+  expect(screen.queryByTestId('patient-add-caregiver-dialog')).not.toBeInTheDocument()
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏
```suggestion
  expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
```